src/app/server/CommissioningWindowManager.cpp

```diff
@@ -184,13 +184,21 @@ CHIP_ERROR CommissioningWindowManager::OpenCommissioningWindow()
     }
     else
     {
-        uint32_t pinCode;
-        ReturnErrorOnFailure(DeviceLayer::ConfigurationMgr().GetSetupPinCode(pinCode));
-
-        ReturnErrorOnFailure(mPairingSession.WaitForPairing(
-            pinCode, kSpake2p_Iteration_Count,
-            ByteSpan(reinterpret_cast<const uint8_t *>(kSpake2pKeyExchangeSalt), strlen(kSpake2pKeyExchangeSalt)), keyID,
-            Optional<ReliableMessageProtocolConfig>::Value(gDefaultMRPConfig), this));
+        uint32_t iterationCount;
+        uint8_t salt[kPBKDFMaximumSaltLen];
+        size_t saltLen;
+        PASEVerifier verifier;
+        size_t verifierLen;
+
+        ReturnErrorOnFailure(DeviceLayer::ConfigurationMgr().GetSpake2pIterationCount(iterationCount));
+        ReturnErrorOnFailure(DeviceLayer::ConfigurationMgr().GetSpake2pSalt(salt, sizeof(salt), saltLen));
+        ReturnErrorOnFailure(DeviceLayer::ConfigurationMgr().GetSpake2pVerifier(reinterpret_cast<uint8_t *>(&verifier),
```

Comment:
This is assuming things about object layout that may not hold (like no padding).

Reply:
that was addressed. Reading serialized PASE Verifier from the memory and then deserialize it